### PR TITLE
Support for task queues

### PR DESCRIPTION
**Pull request by toudi**
- Added support for task queues (backward compatible)
- gearman module version raised to 2.0.2

### CODE REVIEW

File: django_gearman/decorators.py [hunk 1]
```diff
@@ -1,31 +1,43 @@
...
-
-    def __call__(self, *args, **kwargs):
-        # call function with argument passed by the client only
-        try:
-            arg = args[0].arg
-        except IndexError:
-            arg = None
-        return self.f(arg)
+"""
+We need to change gearman_job decorator into a function which contains
```
Comment (fwenzel) [inline django_gearman/decorators.py]: I don't think you need this docstring/comment in the code (though thanks for explaining why you made this a class decorator)

File: django_gearman/management/commands/gearman_list_tasks.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from django.conf import settings
+from django.core.management.base import NoArgsCommand
+from gearman_worker import Command as Worker
+
+class Command(NoArgsCommand):
+    help = "List all avalible gearman jobs with queues they belong to"
```
Comment (fwenzel) [inline django_gearman/management/commands/gearman_list_tasks.py]: typo "available"

File: django_gearman/management/commands/gearman_worker.py [hunk 3]
```diff
@@ -25,9 +28,17 @@ def handle_noargs(self, **options):
             except ImportError:
                 pass
         if not gm_modules:
-            print "No gearman modules found!"
+            self.stdout.write("No gearman modules found!")
```
Comment (fwenzel) [inline django_gearman/management/commands/gearman_worker.py]: you probably want to use `stderr`

File: django_gearman/models.py [hunk 4]
```diff
@@ -15,42 +27,24 @@ def __init__(self, **kwargs):
...
-        parts = func.partition('.')
-        if parts[2]:
-            app = parts[0]
-            job = parts[2]
-        else:
-            app = ''
-            job = parts[0]
-
-        try:
-            func = settings.GEARMAN_JOB_NAME % {'app': app, 'job': job}
```
Comment (fwenzel) [inline django_gearman/models.py:51]: are gearman tasks not namespaced anymore? What if several applications use the same gearman servers?

**Comment by fwenzel** [comment]
There are some formatting errors etc, but I can help fix those. The task namespacing is the biggest question I have. Once we figure this out, I am happy to land this.

Thanks for your submission!

**Comment by toudi** [comment]
What do i need to do in order to apply changes? Do a local commit, then push and then close this pull request and make another one?

As for the the namespacing i'll write about it in a moment, because it needs a longer explanation ;)

**Comment by toudi** [comment]
> are gearman tasks not namespaced anymore? What if several applications use the same gearman servers?

I believe you meant that several applications use the same task names.

I changed this, because i give the tasks their default name of imported module - i believe that i shouldn't delete those lines in this particular pull, but they need to be deleted in the other one (the one which introduces individual task names).

The reason for this method is that i tried to use Your namespacing scheme and i very shortly ran into problem - i was importing other gearman jobs inside gearman_jobs.py file and i couldn't execute those imported tasks in any way (they were given the name, but i was unable to execute them).

Anyways, the other patch gives the task it's full import name (like application.foo.bar.function_name) or custom, user-defined name.

**Comment by fwenzel** [comment]
> What do i need to do in order to apply changes?

Just make a new commit to the same branch, then push it to github. It'll automatically show up in this pull request!

**Comment by fwenzel** [comment]
What you say about namespaces makes sense. Please remove the "gearman job name pattern" part from the README file then!

**Pull request merged**